### PR TITLE
Backport Fix PHP 8.2 warnings by casting trim() argument to string

### PR DESCRIPTION
**Pull request by uvaldenaire-opendsi**
Backport of https://github.com/Dolibarr/dolibarr/pull/28162

### CODE REVIEW

File: htdocs/societe/class/societe.class.php [hunk 1]
```diff
@@ -889,7 +889,7 @@ public function create(User $user, $notrigger = 0)
 		if (empty($this->status)) {
 			$this->status = 0;
 		}
-		$this->name = $this->name ?trim($this->name) : trim($this->nom);
+		$this->name = $this->name ? trim($this->name) : trim((string) $this->nom);
```
Comment (rycks) [inline htdocs/societe/class/societe.class.php:892]: you seems to forget a (string) cast into first trim call no ?

**Review by rycks — APPROVED** [review]
ok for me

**Review by lvessiller-opendsi — APPROVED** [review]

**Comment by hregis** [comment]
@uvaldenaire-opendsi

(string) trim(xxxxx)

Is different of

trim((string) xxxx))

?

**Comment by uvaldenaire-opendsi** [comment]
@hregis 
(string) trim(xxxxx) when xxxxx is null will throw a deprecated warning.
trim((string) xxxx)) when xxxx is null won't throw a deprecated warning.

**Pull request merged**